### ISSUE TITLE
Occasionally `oc cluster up` results in a state that responds with EOF for all requests

### ISSUE THREAD

**Issue by benjaminapetersen** [openshift/origin#14890]
Running this command, on Mac:

```bash
$ oc cluster up --version=latest --service-catalog=true --loglevel=1
```
Results in what looks pretty normal:

```bash
-- Checking OpenShift client ...
-- Checking Docker client ...
-- Checking Docker version ...
-- Checking for existing OpenShift container ...
-- Checking for openshift/origin:latest image ...
-- Checking Docker daemon configuration ...
-- Checking for available ports ...
-- Checking type of volume mount ...

   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ...
-- Finding server IP ...

   Using 127.0.0.1 as the server IP
-- Checking service catalog version requirements ...
-- Starting OpenShift container ...

   Creating initial OpenShift configuration
I0626 15:18:20.999518    2887 helper.go:580] Copying OpenShift config to local directory /var/folders/bh/k6l0fj7j4d7792vq0zbhp2sh0000gn/T/openshift-config156233055
I0626 15:18:21.513476    2887 helper.go:604] Reading master config from /var/folders/bh/k6l0fj7j4d7792vq0zbhp2sh0000gn/T/openshift-config156233055/master/master-config.yaml
I0626 15:18:25.415155    2887 helper.go:593] Reading node config from /var/folders/bh/k6l0fj7j4d7792vq0zbhp2sh0000gn/T/openshift-config156233055/node-localhost/node-config.yaml
   Starting OpenShift using container 'origin'
I0626 15:18:27.724894    2887 helper_unix.go:73] Started socat with pid: 2892
   Waiting for API server to start listening
   OpenShift server started
-- Adding default OAuthClient redirect URIs ...
-- Installing registry ...
-- Installing router ...
-- Importing image streams ...
-- Importing templates ...
-- Installing service catalog ...
-- Login to server ...

   Login successful.

   You don't have any projects. You can try to create a new project, by running

        new-project <projectname>

-- Creating initial project "myproject" ...
-- Removing temporary directory ...
-- Checking container networking ...
-- Server Information ...

   OpenShift server started.

   The server is accessible via web console at:
       https://127.0.0.1:8443

   You are logged in as:
       User:     developer

   To login as administrator:
       oc login -u system:admin

-- Service Catalog Instructions ...

   In order to enable access to the Template Service Broker for use with the Service Catalog, you must first grant unauthenticated access to the template service broker api.

   WARNING: Enabling this access allows anyone who can see your cluster api server to provision templates within your cluster, impersonating any user in the cluster (including administrators).  This can be used to gain full administrative access to your cluster.  Do not allow this access unless you fully understand the implications.  To enable unauthenticated access to the template service broker api, run the following command as cluster admin:

   oc adm policy add-cluster-role-to-group system:openshift:templateservicebroker-client system:unauthenticated system:authenticated

   WARNING: Running the above command allows unauthenticated users to access and potentially exploit your cluster.
```

But when attempting to login with the `system:admin` account:
![screen shot 2017-06-26 at 3 26 49 pm](https://user-images.githubusercontent.com/280512/27556516-ea7883ac-5a83-11e7-8efd-adb6217d2e2e.png)
![screen shot 2017-06-26 at 3 38 27 pm](https://user-images.githubusercontent.com/280512/27556900-8d21513c-5a85-11e7-91fa-42dafa097559.png)

Some extra details:

```bash
$ oc whoami --loglevel=8
I0626 15:27:52.496404    3025 loader.go:354] Config loaded from file /Users/***/.kube/config
I0626 15:27:52.497850    3025 cached_discovery.go:118] returning cached discovery info from /Users/***/.kube/127.0.0.1_8443/servergroups.json
I0626 15:27:52.498846    3025 round_trippers.go:383] GET https://127.0.0.1:8443/oapi/v1/users/~
I0626 15:27:52.498885    3025 round_trippers.go:390] Request Headers:
I0626 15:27:52.498896    3025 round_trippers.go:393]     User-Agent: oc/v3.6.0 (darwin/amd64) openshift/4423ff5
I0626 15:27:52.498913    3025 round_trippers.go:393]     Accept: application/json, */*
I0626 15:27:52.498917    3025 round_trippers.go:393]     Authorization: Bearer ***************
I0626 15:27:52.518916    3025 round_trippers.go:408] Response Status:  in 19 milliseconds
I0626 15:27:52.518934    3025 round_trippers.go:411] Response Headers:
I0626 15:27:52.518972    3025 helpers.go:224] Connection error: Get https://127.0.0.1:8443/oapi/v1/users/~: EOF
F0626 15:27:52.518985    3025 helpers.go:119] Unable to connect to the server: EOF

```
and:

```bash
$ oc login -u system:admin --loglevel=8
I0626 15:29:13.581116    3082 loader.go:354] Config loaded from file /Users/***/.kube/config
I0626 15:29:13.583081    3082 round_trippers.go:383] HEAD https://127.0.0.1:8443/
I0626 15:29:13.583133    3082 round_trippers.go:390] Request Headers:
I0626 15:29:13.655339    3082 round_trippers.go:408] Response Status:  in 72 milliseconds
I0626 15:29:13.655369    3082 round_trippers.go:411] Response Headers:
F0626 15:29:13.655645    3082 helpers.go:119] error: EOF
```

Version:

$ oc version
oc v3.6.0-alpha.2+4423ff5-468-dirty
kubernetes v1.6.1+5115d708d7
features: Basic-Auth

Server https://127.0.0.1:8443
kubernetes v1.6.1+5115d708d7

May be related to #14842.

**Comment by pweil-** [comment]
@benjaminapetersen is this only if you use the service catalog flag?  Anything fishy in the master logs?

**Comment by bparees** [comment]
@benjaminapetersen I assume this is exactly https://github.com/openshift/origin/issues/14842 again, is anything different about it?  (can you access the web console in this case)?

**Comment by bparees** [comment]
hm, I see you weren't reusing a config in this case.

@deads2k does the aggregator issue fixed by https://github.com/openshift/origin/pull/14881 also affect clean starts?

@benjaminapetersen can you get the logs from the origin container?

**Comment by deads2k** [comment]
> @deads2k does the aggregator issue fixed by #14881 also affect clean starts?

Uh, maybe.  It would be an unusual race though.

**Comment by bparees** [comment]
@benjaminapetersen are you still seeing this one?

**Comment by benjaminapetersen** [comment]
@bparees, @pweil- sorry, been on vacation a couple weeks. I'll report back here if I see it pop up again as of today.

**Comment by jorgemoralespou** [comment]
I get the EOF error when I can not connect to the server. 

e.g. using oc-cluster 3.6.0-rc.0 and --service-catalog:

````
oc adm policy add-cluster-role-to-group system:openshift:templateservicebroker-client system:unauthenticated system:authenticated --as=cluster-admin --loglevel 9
I0718 13:02:16.762417   47259 loader.go:354] Config loaded from file /Users/jmorales/.kube/config
I0718 13:02:16.763658   47259 cached_discovery.go:118] returning cached discovery info from /Users/jmorales/.kube/127.0.0.1_8443/servergroups.json
I0718 13:02:16.764356   47259 round_trippers.go:386] curl -k -v -XGET  -H "Impersonate-User: cluster-admin" -H "User-Agent: oc/v3.6.0 (darwin/amd64) openshift/c5e5df2" -H "Accept: application/json, */*" https://127.0.0.1:8443/oapi/v1/clusterpolicybindings/:default
I0718 13:02:16.784262   47259 round_trippers.go:405] GET https://127.0.0.1:8443/oapi/v1/clusterpolicybindings/:default  in 19 milliseconds
I0718 13:02:16.784299   47259 round_trippers.go:411] Response Headers:
I0718 13:02:16.784357   47259 helpers.go:223] Connection error: Get https://127.0.0.1:8443/oapi/v1/clusterpolicybindings/:default: EOF
F0718 13:02:16.784374   47259 helpers.go:119] Unable to connect to the server: EOF
````

**Comment by benjaminapetersen** [comment]
I got an error this morning again, same `getsockopt`, following `oc cluster up --use-existing-config`:

```
$ oc login -u system:admin
error: dial tcp 127.0.0.1:8443: getsockopt: connection refused - verify you have provided the correct host and port and that the server is currently running.
```
This time doing `oc cluster down` & `oc cluster up --use-existing-config` again worked.  Previously I had to nuke the env & start fresh.

**Comment by bparees** [comment]
seems like there are still some api-aggregator kinks, @deads2k

**Comment by bparees** [comment]
(possibly a race condition given that cluster down and then up a second time w/ the same config again, worked)

**Comment by bparees** [comment]
@benjaminapetersen seems like @jorgemoralespou is seeing it even w/o reusing a config.  Have you hit it w/o reusing a config?

**Comment by benjaminapetersen** [comment]
I have not (yet).

**Comment by deads2k** [comment]
Can you gather server logs?

**Comment by benjaminapetersen** [comment]
The next time I hit it I'll get you some.

**Comment by bparees** [comment]
@benjaminapetersen are you still seeing this issue?  if not, can we close?

**Comment by benjaminapetersen** [comment]
I think we are good.  Closing.

**Comment by vivekrgi** [comment]
i can not use any oc commands on my windows machine. getting the below error
runtime: panic before malloc heap initialized
fatal error: runtime: cannot reserve arena virtual address space

runtime stack:
runtime.throw(0x2242c12)
        /usr/lib/golang/src/runtime/panic.go:491 +0x83 fp=0xdfe9c sp=0xdfe84
runtime.mallocinit()
        /usr/lib/golang/src/runtime/malloc.c:223 +0xef fp=0xdfed8 sp=0xdfe9c
runtime.schedinit()
        /usr/lib/golang/src/runtime/proc.c:137 +0x3a fp=0xdfef0 sp=0xdfed8
runtime.rt0_go(0xdff88, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /usr/lib/golang/src/runtime/asm_386.s:100 +0xf9 fp=0xdfef4 sp=0xdfef0

even just running oc version gives this error. could someone pls help.. it was working fine 2 days back and all of sudden getting this error.

**Comment by csrwng** [comment]
@vivekrgi sounds like something is taking up your RAM
https://stackoverflow.com/questions/20234919/go-runtimepanic-before-malloc-heap-initialized

**Comment by vivekrgi** [comment]
Even I thought the same but I have 3GB free memory

**Comment by TiggySravan** [comment]
> > @deads2k does the aggregator issue fixed by #14881 also affect clean starts?
> 
> Uh, maybe. It would be an unusual race though.

wqq
